docs/src/cookbooks/cookbook.md

```diff
@@ -561,3 +561,50 @@ ChiselStage.emitVerilog(new CoercedRegConnect {
   dontTouch(monitor)
 })
 ```
+
+### How do I minimize the number of bits used in an output vector?
+
+Use inferred with and a sequence instead of a Vec():
+
+Consider:
+
+```scala mdoc:silent:reset
+import chisel3._
+
+// Count the number of set bits up to and including each bit position
+class CountBits(width: Int) extends Module {
+  val bits = IO(Input(UInt(width.W)))
+  val countSequence = Seq.tabulate(width)(i => IO(Output(UInt())))
+  val countVector = IO(Output(Vec(width, UInt())))
+  (countSequence zip (0 until bits.asBools.length).map { i =>
+    VecInit(bits.asBools.slice(0, i + 1)).count { b: Bool =>
+      b
+    }
+  }).foreach {
+    case (a, b) =>
+      a := b
+  }
```

Comment:
```suggestion
  countSequence.zipWithIndex.foreach { case (port, i) =>
    port := util.PopCount(bits(i, 0))
  }
```

We can implement this a little more simply. Note that `PopCount` is effectively implemented in the same way and gives the same Verilog (although my change here changes the names slightly, most of the emporaries get prefixed by `countSequence`.